packages/bibliography/src/main.rs

```diff
@@ -0,0 +1,358 @@
+                    "to": ["any"],
+                    "arguments": [],
+                    "description": "Do not use this module",
+                    "variables": {
+                        "inline_citation_labels": {"type": "set", "access": "read"}
+                    }
+                },
+                {
+                    "from": "bibliography",
+                    "to": ["any"],
+                    "description": "Inserts a bibliography into the document. \
+                    This module needs to exist for [cite]s to work properly. \
+                    The bibliography may be read from the BibLaTeX or Hayagriva YAML format, \
+                    from either the body of this module or a file passed in as the \"file\" argument.",
+                    "arguments": [
+                        {"name": "style", "default": "IEEE", "type": ["IEEE", "APA", "MLA", "CMoS"], "description": "The style to have the bibliography in"},
+                        {"name": "file", "default": "", "description": "A file containing BibLaTeX or Hayagriva YAML with the bibliography"},
+                        {"name": "visibility", "default": "shown", "type": ["shown", "hidden"], "description": "Whether the bibliography is 'shown' or 'hidden'. \
+                        Note that a [bibliography] must exist for [cite]s to work, and if you don't want a bibliography in your document, you can set this argument to 'hidden'."}
+                    ],
+                    "variables": {
+                        "inline_citations": {"type": "list", "access": "read"},
+                        "inline_citation_labels": {"type": "set", "access": "add"},
+                        "imports": {"type": "set", "access": "add"}
+                    }
+                }
+            ]
+            }
+        )
+    );
+}
+
+// Cyclic dependencies.
+// Consider this document
+// [cite] a
+// [cite] b
+// If cite is both unknown_content and push access to the list, it means a must happen before
+// b (due to the order they appear in the document, list access is granular).
+// a would expand to one [list-push] and something else, but then we have a [list-push] which is
+// above b, which also requires list pushing to the same list. the [list-push] from a must thus
+// occur before b, but also, b is unknown-content and so it must happen before a. Thus, we have
+// a cyclic dependency.
+
+fn transform(from: &str, to: &str) {
+    let input: Value = {
+        let mut buffer = String::new();
+        io::stdin().read_to_string(&mut buffer).unwrap();
+        serde_json::from_str(&buffer).unwrap()
+    };
+
+    match from {
+        "cite" => transform_cite(to, &input),
+        "cite-internal-do-not-use" => transform_cite_label(to, &input),
+        "bibliography" => transform_bibliography(to, &input),
+        other => {
+            eprintln!("Package does not support {other}");
+        }
+    }
+}
+
+// The name Citation clashes with Hayagriva
```

Comment:
I think we can remove the comment since this name is fine too : ). Or perhaps import `Citation as HayagrivaCitation`
```suggestion
```